### PR TITLE
KOGITO-828: RuleSetNodeVisitor: refactor duplicated code 

### PR DESCRIPTION
**Pull request by evacchi**
Improved code everywhere in rule unit handling. All tests should still pass.

I actually spotted a bug in the way "singleton" process variables are handled. Because currently we are using a "collection" data source ("DataStore"), the variable may be set to a "previous" value (ordering is not guaranteed). I have fixed this in the test case by first removing the contents of the variable and then setting it (see Generated.drl)

This will be fixed when we introduce "singleton" data sources (KOGITO-892)

### CODE REVIEW

File: kogito-codegen/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 <!--  <logger name="org.kie.kogito.codegen" level="debug"/>-->
 
-  <root level="info">
+  <root level="debug">
```
Comment (mswiderski) [inline kogito-codegen/src/test/resources/logback-test.xml]: maybe let's get back to info

File: kogito-codegen/src/test/resources/ruletask/Generated.drl [hunk 2]
```diff
@@ -13,8 +13,10 @@ end
 
 rule singlePerson_add_note when
     p: /singlePerson[ age == 50 ]
+    s: /singleString[ this == "hello" ]
 then
     p.setAdult(true);
+    singleString.remove(s);
```
Comment (mswiderski) [inline kogito-codegen/src/test/resources/ruletask/Generated.drl]: what's the reason for this change?

Comment (evacchi) [reply kogito-codegen/src/test/resources/ruletask/Generated.drl]: description above
> Because currently we are using a "collection" data source ("DataStore"), the variable may be set to a "previous" value (ordering is not guaranteed). I have fixed this in the test case by first removing the contents of the variable and then setting it (see Generated.drl)

File: kogito-codegen/src/test/resources/logback-test.xml (hunk 1 above)
Comment (evacchi) [reply kogito-codegen/src/test/resources/logback-test.xml]: thanks

File: kogito-codegen/src/test/resources/ruletask/Generated.drl (hunk 2 above)
Comment (evacchi) [reply kogito-codegen/src/test/resources/ruletask/Generated.drl]: basically, previous version was ordering dependent and the refactoring showed that it was broken (it was sometimes green but most times red)
this will be fixed in KOGITO-892

Comment (mswiderski) [reply kogito-codegen/src/test/resources/ruletask/Generated.drl]: ok, thanks for explanation
